### PR TITLE
Unit tests for BitVector

### PR DESCRIPTION
**Pull request by 4tXJ7f**

### CODE REVIEW

File: test/unit/util/bitvector_black.h [hunk 1]
```diff
@@ -26,30 +26,147 @@ class BitVectorBlack : public CxxTest::TestSuite {
...
+   TS_ASSERT_EQUALS(12u, b4.getSize());
+   TS_ASSERT_EQUALS("000000011010", b4.toString());
+   TS_ASSERT_EQUALS("26", b4.toString(10));
+   TS_ASSERT_EQUALS("1a", b4.toString(16));
+  }
+
+  void testConversions()
+  {
+    TS_ASSERT_EQUALS(two.toSignedInteger(), Integer(2));
+    TS_ASSERT_EQUALS("1111", negOne.toString());
```
Comment (aniemetz) [inline test/unit/util/bitvector_black.h]: Flip arguments for consistency?
`TS_ASSERT_EQUALS(negOne.toString(), "1111");`

File: test/unit/util/bitvector_black.h [hunk 2]
```diff
@@ -26,30 +26,147 @@ class BitVectorBlack : public CxxTest::TestSuite {
...
+  {
+    TS_ASSERT_EQUALS(one.setBit(1).setBit(2).setBit(3), negOne);
+
+    TS_ASSERT(negOne.isBitSet(3));
+    TS_ASSERT(!two.isBitSet(3));
+
+    TS_ASSERT_EQUALS(one.getValue(), Integer(1));
+    TS_ASSERT_EQUALS(zero.isPow2(), 0);
+    TS_ASSERT_EQUALS(one.isPow2(), 1);
+    TS_ASSERT_EQUALS(two.isPow2(), 2);
```
Comment (aniemetz) [inline test/unit/util/bitvector_black.h]: Add a test for a value that is not a power of two (negOne) ?

File: test/unit/util/bitvector_black.h [hunk 3]
```diff
@@ -26,30 +26,147 @@ class BitVectorBlack : public CxxTest::TestSuite {
...
+  }
+
+  void testConcatExtract()
+  {
+    BitVector b = one.concat(zero);
+    TS_ASSERT_EQUALS(b.toString(), "00010000");
+    TS_ASSERT_EQUALS(b.extract(7, 4), one);
+    TS_ASSERT_EQUALS(b.concat(BitVector()), b);
```
Comment (aniemetz) [inline test/unit/util/bitvector_black.h]: I'm missing error tests here, e.g., when `high` and `low` of extract do not match specifications:
`high < low`
`high >= d_size`
What happens if you do a `BitVector().extract(2, 1)` ?

File: test/unit/util/bitvector_black.h [hunk 4]
```diff
@@ -26,30 +26,147 @@ class BitVectorBlack : public CxxTest::TestSuite {
...
+  }
+
+  void testComparisons()
+  {
+    TS_ASSERT_DIFFERS(zero, one);
+    TS_ASSERT(negOne > zero);
+    TS_ASSERT(negOne >= zero);
+    TS_ASSERT(negOne >= negOne);
```
Comment (aniemetz) [inline test/unit/util/bitvector_black.h]: What about `negOne == negOne` ?

File: test/unit/util/bitvector_black.h (hunk 3 above)
Comment (4tXJ7f) [reply test/unit/util/bitvector_black.h]: Added the error tests and error checks in BitVector::extract.

**Pull request merged**